### ISSUE TITLE
Support multiple lines in graph3d

### ISSUE THREAD

**Issue by nzjrs** [almende/vis#2840]
It would be great if visjs supported multiple lines on the plot. Even better if these could be updated in a way similar to how one can do it with multiple points in a scatter plot (i.e. updating points with a certain ID).

In the best case that would enable easy updating of 3d lines in a similar manner to http://visjs.org/examples/graph3d/06_moving_dots.html

this would be a feature unique to visjs, nothing else supports this.

**Comment by Tooa** [comment]
Not sure, but I think @wimrijnders is working towards this goal.

**Comment by wimrijnders** [comment]
OK, so this *is* a graph3d issue. I see the tag now.

Hi `nzjrs`,

What you describe is indeed what I am working on; not just for line plots, but for any available graph type in `graph3d`. The things is, it's not trivial, but I'm persevering.  As it is, I'm the process of adding in the changes piecewise so that they are reviewable.

So I request you have patience; it's going to be possible in due time. Not tomorrow, but soon enough.

@yotamberk this is also a subset of #1913, which I take to be leading for my changes.

**Comment by nzjrs** [comment]
Hi @wimrijnders 

Thanks for your work on this. Let me know if I can help. It would be great if these improvements retain the ability to modify the line once it has been drawn - so continuing the rich support for the visjs DataSet. I appreciate this is probably the most challenging part however. Thanks once again.

**Comment by wimrijnders** [comment]
@nzjrs, if by 'modification' you mean moving the graph points as in the [moving dots example](http://visjs.org/examples/graph3d/06_moving_dots.html), yes, that will indeed be possible, on a per-graph basis. I'm actually really looking forward to make a demo for that!

**Comment by nzjrs** [comment]
@wimrijnders yeah, more or less. However that example uses 'filter'. In my application I just change the values for the points with the given ID and the DataSet data binding takes care of ensuring the graph is updated.

**Comment by xogeny** [comment]
Are there any updates on this?

**Comment by nzjrs** [comment]
Hi @wimrijnders apologies for the intrusion.

Just wondering how the various plumbing to enable this feature is progressing and if you are still working on it?
